File: backend/app/Models/Production.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Production extends Model
{
    protected $fillable = [
        'producteur_id',
        'code_tracabilite',
        'type_culture',
        'superficie',
        'date_recolte',
        'quantite_estimee',
        'quantite_reelle',
        'statut',
    ];

    protected static function booted(): void
    {
        static::creating(function (Production $production) {
            if (empty($production->code_tracabilite)) {
                $production->code_tracabilite = 'LOT-' . now()->format('Y') . '-' . strtoupper(Str::random(6));
            }
        });
    }

    public function producteur()
    {
        return $this->belongsTo(Producteur::class, 'producteur_id');
    }

    public function stocks()
    {
        return $this->hasMany(Stock::class, 'production_id');
    }
}

File: mobile-app/lib/models/production.dart
class Production {
  final int id;
  final String codeTracabilite;
  final String? typeCulture;
  final double? superficie;
  final String? dateRecolte;
  final double? quantiteEstimee;
  final double? quantiteReelle;
  final String statut;

  const Production({
    required this.id,
    required this.codeTracabilite,
    this.typeCulture,
    this.superficie,
    this.dateRecolte,
    this.quantiteEstimee,
    this.quantiteReelle,
    required this.statut,
  });

  factory Production.fromJson(Map<String, dynamic> j) => Production(
        id: j['id'],
        codeTracabilite: j['code_tracabilite'],
        typeCulture: j['type_culture'],
        superficie: (j['superficie'] as num?)?.toDouble(),
        dateRecolte: j['date_recolte'],
        quantiteEstimee: (j['quantite_estimee'] as num?)?.toDouble(),
        quantiteReelle: (j['quantite_reelle'] as num?)?.toDouble(),
        statut: j['statut'] ?? 'en_attente',
      );
}
